scripts/gtcadis.py

```diff
@@ -0,0 +1,233 @@
+
+    xmesh = cfg['xmesh']
+    xints = cfg['xints']
+    ymesh = cfg['ymesh']
+    yints = cfg['yints']
+    zmesh = cfg['zmesh']
+    zints = cfg['zints']
+
+    # Create structured mesh
+    sc = [np.linspace(float(origin_x), float(xmesh), float(xints) + 1),
+          np.linspace(float(origin_y), float(ymesh), float(yints) + 1),
+          np.linspace(float(origin_z), float(zmesh), float(zints) + 1)]
+    m = Mesh(structured=True, structured_coords=sc)
+    m.mesh.save("blank_mesh.h5m")
+
+    # Generate 42 photon energy bins [eV]
+    #  First bin has been replaced with 1 for log interpolation
+    photon_bins = np.array([1e-6, 0.01, 0.02, 0.03, 0.045, 0.06, 0.07, 0.075, 0.1, 0.15,
+                            0.2, 0.3, 0.4, 0.45, 0.51, 0.512, 0.6, 0.7, 0.8, 1, 1.33, 1.34,
+                            1.5, 1.66, 2, 2.5, 3, 3.5, 4, 4.5, 5, 5.5, 6, 6.5, 7, 7.5, 8, 10,
+                            12, 14, 20, 30, 50])
+    # ICRP 74 flux-to-dose conversion factors in pico-Sv/s per photon flux
+    de = np.array([0.01, 0.015, 0.02, 0.03, 0.04, 0.05, 0.06, 0.07, 0.08, 0.1, 0.15, 0.2, 0.3,
+                   0.4, 0.5, 0.6, 0.8, 1, 2, 4, 6, 8, 10])
+    df = np.array([0.0485, 0.1254, 0.205, 0.2999, 0.3381, 0.3572, 0.378, 0.4066, 0.4399, 0.5172,
+                   0.7523, 1.0041, 1.5083, 1.9958, 2.4657, 2.9082, 3.7269, 4.4834, 7.4896,
+                   12.0153, 15.9873, 19.9191, 23.76])
+    # Convert to Sv/s per photon FLUX 
+    pico = 1.0e-12 
+    df = df * pico 
+    # Convert pointwise data to group data for log interpolation
+    photon_spectrum = pointwise_collapse(
+        photon_bins, de, df, logx=True, logy=True)
+    #  Anything below 0.01 MeV should be assigned the DF value of 0.01 MeV
+    photon_spectrum[0] = df[0]
+    # Total number of groups is 217 (42 photon + 175 neutron)
+    spectra = [np.append(photon_spectrum, np.zeros(175))]
+    # The spectrum is normalized by PyNE, so we need to mutliply by the sum of
+    # intensities in the spectrum.
+    # Additionally, we divide by the volume of the source cell in order to get
+    # source density.
+    intensities = [np.sum(spectra) / src_vol]
+
+    # Load geometry into DAGMC
+    load(geom)
+    # Generate isotropic photon volume source
+    source, dg = isotropic_vol_source(geom, m, cells, spectra, intensities)
+
+    # PARTISN input
+    ngroup = 217  # total number of energy groups
+    cards = _cards(source)  # block 1, 3, 5 input values
+    names_dict = _names_dict()  # dictionary of isotopes (PyNE nucids to bxslib names)
+
+    write_partisn_input(
+        m,
+        geom,
+        ngroup,
+        cards=cards,
+        dg=dg,
+        names_dict=names_dict,
+        data_hdf5path="/materials",
+        nuc_hdf5path="/nucid",
+        fine_per_coarse=1)
+
+
+def main():
+    """ This function manages the setup and steps 1-5 for the GT-CADIS workflow.
+    """
+
+    with open(config_filename, 'r') as f:
+        cfg = yaml.load(f)
```

Comment:
Shouldn't the above two lines come later after setup is run?. This should give an error as the config.yml file hasn't been written yet till 'setup' is run. I guess it should be moved below "if args.command == 'setup': setup()"